### PR TITLE
Script optimization maintenance fixes

### PR DESCRIPTION
**Pull request by vpax**
This PR started off to update script optimization to deal with the new `<<`/`>>` bit-shifting operators, but grew in several ways.

First, it turns out the recent btest additions for vector bit-shifting were broken, but that fact wasn't caught because the test only looked at `stdout` and not `stderr`.  I updated all the `language/` btests that use `test_case()` to include `.stderr` in their baselines.

Second, it turns out that some of the recent error checks for bit-shifting were not in fact needed (due to previous type-checking ruling them out), so I simplified those.

Third, I changed a number of btests that leverage a set of `local` variables for low-level tests to instead use `global` variables.  This is necessary because for locals, script optimization is smart enough to use constant propagation and compute many of the results at compile-time rather than run-time, meaning the run-time correctness was in fact not being tested.

Lastly, this PR brings ZAM up to date (the `-a zam` btests complete successfully), and includes a good amount of `-O gen-C++` maintenance, though there are still a number of pending issues that I'll tackle when I have more time.  This latter also included updating a number of btests to explicitly not run when using C++ scripts, and some updates to the associated maintenance scripts.

### CODE REVIEW

**Review by timwoj — APPROVED** [review]

File: testing/btest/language/addr.zeek [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
+global b2: addr = [::255.255.255.255];
+global b3: addr = [::ffff:ffff];
+global b4: addr = [ffff::ffff];
+global b5: addr = [0000:0000:0000:0000:0000:0000:0000:0000];
+global b6: addr = [aaaa:bbbb:cccc:dddd:eeee:ffff:1111:2222];
+global b7: addr = [AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:1111:2222];
+global b9 = [2001:db8:0:0:0:FFFF:192.168.0.5];
+
+# IPv4-mapped-IPv6 (internally treated as IPv4)
+global c1: addr = [::ffff:1.2.3.4];
```
[the remark this answers is not in the document]
Comment (vpax) [reply testing/btest/language/addr.zeek:27]: See the penultimate paragraph of the (long-winded :-) PR writeup

File: src/Expr.cc [hunk 2]
```diff
@@ -963,40 +963,27 @@ ValPtr BinaryExpr::Fold(Val* v1, Val* v2) const
 			{
 			if ( is_integral )
 				{
-				if ( i2 < 0 )
-					RuntimeError("left shift by negative value");
-				else if ( i1 < 1 )
+				if ( i1 < 1 )
```
[the remark this answers is not in the document]
Comment (vpax) [reply src/Expr.cc:966]: Good point!  I agree that this should be just for strictly negative.

**Comment by vpax** [comment]
@timwoj the CI failures are for systems that don't support `export -n` to remove an environment variable export.  I'm quite weak on portable `sh` idioms, so am not sure whether there's a blessed way of doing this.  If not then we can instead skip the test if we're doing C++ stuff (like I added for a number of other tests).

File: testing/btest/language/addr.zeek (hunk 1 above)
[the remark this answers is not in the document]
Comment (timwoj) [reply testing/btest/language/addr.zeek:27]: Ah yes, reading, my old nemesis.

**Comment by timwoj** [comment]
> the CI failures are for systems that don't support export -n to remove an environment variable export. I'm quite weak on portable sh idioms, so am not sure whether there's a blessed way of doing this.

I usually use `unset`, which appears to be used in other tests as well. See `bifs.rand` and `core.conn-uid`, for example.

**Review by timwoj — APPROVED** [review]

**Pull request merged**